scripts/surgical_robotics_challenge/examples/socket_based_control/socket_based_control.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+# //==============================================================================
+# /*
+#     Software License Agreement (BSD License)
+#     Copyright (c) 2023 Akhil Deo <adeo1@jhu.edu>
+
+
+#     All rights reserved.
+
+#     Redistribution and use in source and binary forms, with or without
+#     modification, are permitted provided that the following conditions
+#     are met:
+
+#     * Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+
+#     * Redistributions in binary form must reproduce the above
+#     copyright notice, this list of conditions and the following
+#     disclaimer in the documentation and/or other materials provided
+#     with the distribution.
+
+#     * Neither the name of authors nor the names of its contributors may
+#     be used to endorse or promote products derived from this software
+#     without specific prior written permission.
+
+#     THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#     "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#     LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+#     FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+#     COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+#     INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+#     BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+#     LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+#     CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+#     LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+#     ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#     POSSIBILITY OF SUCH DAMAGE.
+
+
+#     \author    <adeo1@jhu.edu>
+#     \author    Akhil Deo
+#     \version   1.0
+# */
+# //==============================================================================
+
+from ambf_client import Client
+import rospy
+from PyKDL import Rotation, Frame, Vector
+import socket
+import json
+import time
+from surgical_robotics_challenge.psm_arm import PSM
+from surgical_robotics_challenge.ecm_arm import ECM
+import numpy as np
+import sys
+import signal
+
+
+UDP_IP = socket.gethostbyname(socket.gethostname())
+UDP_PORT = 8080
+sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+sock.bind(("", UDP_PORT))
+
+_client = Client()
+_client.connect()
+print(_client.get_obj_names())
+w = _client.get_world_handle()
+w.reset_bodies()
+psm1 = PSM(_client, 'psm1')
```

Comment:
The PSM and ECM class can take in the SimulationManager object that I mentioned above